panther_battery/src/adc_node.py

```diff
@@ -240,19 +250,35 @@ def _count_volt_mean(self, label: Union[rospy.Publisher, str], new_val: float) -
         # Updates the average by adding the newest and removing the oldest component of mean value,
         # in order to avoid recalculating the entire sum every time.
         self._V_bat_mean[label] += (new_val - self._V_bat_hist[label][0]) / self._volt_mean_length
-        
+
         self._V_bat_hist[label].pop(0)
         self._V_bat_hist[label].append(new_val)
 
         return self._V_bat_mean[label]
 
+    def _count_curr_mean(self, label: rospy.Publisher, new_val: float) -> float:
+        # Updates the average by adding the newest and removing the oldest component of mean value,
+        # in order to avoid recalculating the entire sum every time.
+        self._I_bat_charge_mean[label] += (
+            new_val - self._I_bat_charge_hist[label][0]
+        ) / self._curr_mean_length
+
+        self._I_bat_charge_hist[label].pop(0)
+        self._I_bat_charge_hist[label].append(new_val)
+
+        return self._I_bat_charge_mean[label]
+
```

Comment:
This method looks exactly as `_count_volt_mean`. Maybe this should be one method with additional arguments?